Sources/NIO/BaseSocket.swift

```diff
@@ -58,15 +58,39 @@ import CNIOWindows
 #endif
 
 /// A Registration on a `Selector`, which is interested in an `SelectorEventSet`.
-/// `sequenceIdentifier` is used by the event notification backends (kqueue, epoll, ...)
+/// `registrationID` is used by the event notification backends (kqueue, epoll, ...)
 /// to mark events to allow for filtering of received return values to not be delivered to a
 /// new `Registration` instance that receives the same file descriptor. Ok if it wraps.
 /// Needed for i.e. testWeDoNotDeliverEventsForPreviouslyClosedChannels to succeed.
-typealias RegistrationSequenceIdentifier = UInt32
+@usableFromInline struct SelectorRegistrationID: Equatable {
+    @usableFromInline var _id: UInt32
```

Comment:
Do we need `_id` here? Doesn't `@usableFromInline internal private(set) var rawValue: UInt32` achieve the same?

Reply:
@glbrntt no, this doesn't work, we do write to it